dependencies/src/test/java/com/google/cloud/tools/opensource/classpath/ClassDumperTest.java

```diff
@@ -142,4 +77,57 @@ public void testMethodDescriptorToClass_primitiveTypes() {
         .inOrder();
   }
 
+  @Test
+  public void testMethodDefinitionExists_arrayType() throws ClassNotFoundException {
+    FullyQualifiedMethodSignature checkArgumentMethod =
+        new FullyQualifiedMethodSignature(
+            "com.google.common.base.Preconditions", "checkArgument", "(ZLjava/lang/Object;)V");
+    ClassDumper classDumper = ClassDumper.create(ImmutableList.of());
+
+    Assert.assertTrue(classDumper.methodDefinitionExists(checkArgumentMethod));
+  }
+
+  @Test
+  public void testMethodDefinitionExists_constructorInAbstractClass()
+      throws ClassNotFoundException {
+    ClassDumper classDumper = ClassDumper.create(ImmutableList.of());
+    FullyQualifiedMethodSignature constructorInAbstract =
+        new FullyQualifiedMethodSignature(
+            "com.google.common.collect.LinkedHashMultimapGwtSerializationDependencies",
+            "<init>",
+            "(Ljava/util/Map;)V");
+
+    Assert.assertTrue(classDumper.methodDefinitionExists(constructorInAbstract));
+  }
+
+  @Test
+  public void testListInnerClasses() throws IOException {
+    InputStream classFileInputStream = URLClassLoader.getSystemResourceAsStream(
+        EXAMPLE_CLASS_FILE);
+    ClassParser parser = new ClassParser(classFileInputStream, EXAMPLE_CLASS_FILE);
+    JavaClass javaClass = parser.parse();
+
+    Set<String> innerClassNames = ClassDumper.listInnerClassNames(javaClass);
+    Truth.assertThat(innerClassNames).containsExactly(
+        "com.google.firestore.v1beta1.FirestoreGrpc$FirestoreFutureStub",
+        "com.google.firestore.v1beta1.FirestoreGrpc$FirestoreMethodDescriptorSupplier",
+        "com.google.firestore.v1beta1.FirestoreGrpc$1",
+        "com.google.firestore.v1beta1.FirestoreGrpc$MethodHandlers",
+        "com.google.firestore.v1beta1.FirestoreGrpc$FirestoreStub",
+        "com.google.firestore.v1beta1.FirestoreGrpc$FirestoreBaseDescriptorSupplier",
+        "com.google.firestore.v1beta1.FirestoreGrpc$FirestoreBlockingStub",
+        "com.google.firestore.v1beta1.FirestoreGrpc$FirestoreImplBase",
+        "com.google.firestore.v1beta1.FirestoreGrpc$FirestoreFileDescriptorSupplier"
+    );
+  }
+
+  @Test
+  public void testCreationInvalidInput() {
+    try {
+      ClassDumper.create(ImmutableList.of(Paths.get("")));
+      Assert.fail("Empty path should generate IOException");
```

Comment:
but you're catching a RuntimeException, not an IOException?

Reply:
Updated message